Goal: Support non Linux ports

Fixes: 266693

Status wrt upstream: Forwarded, will be in 3.0.26

Note: Upstream would probably welcome patches allowing to build on new architectures. 
      As this patch is the most often broken by new upstream releases, everybody would benefit from it.

Index: samba-3.3.0rc2/source/configure.in
===================================================================
--- samba-3.3.0rc2.orig/source/configure.in
+++ samba-3.3.0rc2/source/configure.in
@@ -553,6 +553,15 @@
       fi
     ;;
 
+# Systems with LFS support.
+#
+    gnu* | k*bsd*-gnu)
+	CPPFLAGS="-D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE $CPPFLAGS"
+	AC_DEFINE(_LARGEFILE64_SOURCE, 1, [Whether to enable large file support])
+	AC_DEFINE(_FILE_OFFSET_BITS, 64, [File offset bits])
+	AC_DEFINE(_GNU_SOURCE, 1, [Whether to use GNU libc extensions])
+	;;
+
 # Tests for linux LFS support. Need kernel 2.4 and glibc2.2 or greater support.
 #
     *linux*)
@@ -1222,7 +1231,7 @@
 #
 #
 case "$host_os" in
-    *linux*)
+    linux*-gnu* | gnu* | k*bsd*-gnu)
        # glibc <= 2.3.2 has a broken getgrouplist
        AC_TRY_RUN([
 #include <unistd.h>
@@ -1656,7 +1665,10 @@
 
   # and these are for particular systems
   case "$host_os" in
-		*linux*)   AC_DEFINE(LINUX,1,[Whether the host os is linux])
+		linux*-gnu* | gnu* | k*bsd*-gnu)
+			case "$host_os" in linux*)
+				AC_DEFINE(LINUX,1,[Whether the host os is linux]) ;;
+			esac
 			BLDSHARED="true"
 			if test "${ac_cv_gnu_ld_no_default_allow_shlib_undefined}" = "yes"; then
 				LDSHFLAGS="-shared -Wl,-Bsymbolic -Wl,--allow-shlib-undefined"
@@ -5375,7 +5387,7 @@
 	AC_MSG_RESULT(yes);
 
 	case "$host_os" in
-	*linux*)
+	linux*-gnu* | gnu* | k*bsd*-gnu)
 		AC_CACHE_CHECK([for linux sendfile64 support],samba_cv_HAVE_SENDFILE64,[
 		AC_TRY_LINK([#include <sys/sendfile.h>],
 [\
@@ -5701,11 +5713,11 @@
 WINBIND_NSS_PTHREAD=""
 
 case "$host_os" in
-	*linux*)
+	linux*-gnu* | gnu* | k*bsd*-gnu)
 		NSSSONAMEVERSIONSUFFIX=".2"
 		WINBIND_NSS_EXTRA_OBJS="nsswitch/winbind_nss_linux.o"
 		;;
-	*freebsd[[5-9]]*)
+	freebsd5*|*freebsd[[6-9]]*)
 		# FreeBSD winbind client is implemented as a wrapper around
 		# the Linux version.
 		NSSSONAMEVERSIONSUFFIX=".1"
